examples/vision/xray_classification_with_tpus.py

```diff
@@ -0,0 +1,489 @@
+would prefer to have a less extreme division between the training and the validation set.
+We will append the validation files and create a new split that resembes the standard
+80:20 division instead.
+"""
+
+filenames = tf.io.gfile.glob(str(GCS_PATH + "/chest_xray/train/*/*"))
+filenames.extend(tf.io.gfile.glob(str(GCS_PATH + "/chest_xray/val/*/*")))
+
+train_filenames, val_filenames = train_test_split(filenames, test_size=0.2)
+
+"""
+Run the following cell to see how many healthy/normal chest X-rays we have and how many
+pneumonia chest X-rays we have.
+"""
+
+COUNT_NORMAL = len([filename for filename in train_filenames if "NORMAL" in filename])
+print("Normal images count in training set: " + str(COUNT_NORMAL))
+
+COUNT_PNEUMONIA = len(
+    [filename for filename in train_filenames if "PNEUMONIA" in filename]
+)
+print("Pneumonia images count in training set: " + str(COUNT_PNEUMONIA))
+
+"""
+Notice that the there are way more images that are classified as pneumonia than normal.
+This shows that we have a imbalance in our data. We will correct for this imbalance later
+on in our notebook.
+"""
+
+train_list_ds = tf.data.Dataset.from_tensor_slices(train_filenames)
+val_list_ds = tf.data.Dataset.from_tensor_slices(val_filenames)
+
+for f in train_list_ds.take(5):
+    print(f.numpy())
+
+"""
+Run the following cell to see how many images we have in our training dataset and how
+many images we have in our validation set. Verify that the ratio of images is 80:20.
+"""
+
+TRAIN_IMG_COUNT = tf.data.experimental.cardinality(train_list_ds).numpy()
+print("Training images count: " + str(TRAIN_IMG_COUNT))
+
+VAL_IMG_COUNT = tf.data.experimental.cardinality(val_list_ds).numpy()
+print("Validating images count: " + str(VAL_IMG_COUNT))
+
+"""
+As expected, we have two labels for our images.
+"""
+
+CLASS_NAMES = np.array(
+    [
+        str(tf.strings.split(item, os.path.sep)[-1].numpy())[2:-1]
+        for item in tf.io.gfile.glob(str(GCS_PATH + "/chest_xray/train/*"))
+    ]
+)
+CLASS_NAMES
+
+"""
+Currently our dataset is just a list of filenames. We want to map each filename to the
+corresponding (image, label) pair. The following methods will help us do that.
+
+As we only have two labels, we will rewrite the label so that `1` or `True` indicates
+pneumonia and `0` or `False` indicates normal.
+"""
+
+
+def get_label(file_path):
+    # convert the path to a list of path components
+    parts = tf.strings.split(file_path, os.path.sep)
+    # The second to last is the class-directory
+    return parts[-2] == "PNEUMONIA"
+
+
+"""
+The images originally have values that range from [0, 255]. CNNs work better with smaller
+numbers so we will scale this down.
+"""
+
+
+def decode_img(img):
+    # convert the compressed string to a 3D uint8 tensor
+    img = tf.image.decode_jpeg(img, channels=3)
+    # Use `convert_image_dtype` to convert to floats in the [0,1] range.
```

Comment:
Prefer starting the model with a `Rescaling` layer: https://keras.io/api/layers/preprocessing_layers/image_preprocessing/rescaling/

This makes your model portable.

Reply:
Adding a Rescaling layer changes the dimension so fixed the issue by indexing the image tensor.
```
    inputs = tf.keras.Input(shape=(IMAGE_SIZE[0], IMAGE_SIZE[1], 3)),
    x = tf.keras.layers.experimental.preprocessing.Rescaling(1./255)(inputs)[0]
    x = tf.keras.layers.Conv2D(16, 3, activation='relu', padding='same')(x)
```